### PR TITLE
Add ability to do multiple initial poses

### PR DESCRIPTION
**Pull request by dantwinkler**

### CODE REVIEW

File: amcl/src/amcl_node.cpp [hunk 1]
```diff
@@ -326,16 +330,19 @@ AmclNode::AmclNode() :
...
         first_map_received_(false),
         first_reconfigure_call_(true),
         tdr_("AMCL")
 {
   boost::recursive_mutex::scoped_lock l(configuration_mutex_);
 
   // Grab params off the param server
-  private_nh_.param("use_map_topic", use_map_topic_, false);
-  private_nh_.param("first_map_only", first_map_only_, false);
+  private_nh_.param("use_map_topic", use_map_topic_, true);
```
Comment (grieneis) [inline amcl/src/amcl_node.cpp]: Did you need to change the param defaults to true?  We could set the params in the launch file instead.

Also, you immediately hard code the values to true after getting them as params.

File: amcl/src/amcl_node.cpp [hunk 2]
```diff
@@ -443,10 +450,10 @@ AmclNode::AmclNode() :
   laser_scan_filter_->registerCallback(boost::bind(&AmclNode::laserReceived,
                                                    this, _1));
   initial_pose_sub_ = nh_.subscribe("initialpose", 2, &AmclNode::initialPoseReceived, this);
+  initial_poses_sub_ = nh_.subscribe("initialposes", 2, &AmclNode::initialPosesReceived, this);
 
   if(use_map_topic_) {
-    map_sub_ = nh_.subscribe("map", 1, &AmclNode::mapReceived, this);
-    ROS_INFO("Subscribed to map topic.");
+	map_sub_ = nh_.subscribe("map", 1, &AmclNode::mapReceived, this);		    ROS_INFO("Subscribed to map topic.");
```
Comment (grieneis) [inline amcl/src/amcl_node.cpp]: A new line got killed.

File: amcl/src/amcl_node.cpp [hunk 3]
```diff
@@ -1436,8 +1443,32 @@ AmclNode::initialPoseReceived(const geometry_msgs::PoseWithCovarianceStampedCons
 }
 
 void
+AmclNode::initialPosesReceived(const move_base_msgs::PoseWithCovarianceStampedArrayConstPtr& msg)
+{
+  handleInitialPosesMessage(*msg);
+
+  // Force one no motion update
+  m_force_update = true;
```
Comment (grieneis) [inline amcl/src/amcl_node.cpp]: Is m_force_update a new variable?  Might be better to make it "force_update_".

File: amcl/src/amcl_node.cpp (hunk 1 above)
Comment (dantwinkler) [reply amcl/src/amcl_node.cpp]: I did not mean to change these, was fighting the values not getting updated when running the node standalone in the debugger.  Turns out it was just that a roscore was running in a separate window (param server was caching the values).

Comment (dantwinkler) [reply amcl/src/amcl_node.cpp]: Reverted to original.

File: amcl/src/amcl_node.cpp (hunk 3 above)
Comment (dantwinkler) [reply amcl/src/amcl_node.cpp]: It is not a new variable but I agree that it should be changed to follow coding style.

**Review by grieneis — APPROVED** [review]

**Pull request merged**